server/controllers/routes/admin/index.js

```diff
@@ -9,19 +9,24 @@ const {
   getCohortProjects,
 } = require('./project');
 const getStats = require('./stats');
+const login = require('./login');
 const logout = require('./logout');
+const protectedRoute = require('../../middlewares/auth');
 
-router.post('/cohorts', addCohort);
-
-router.route('/cohorts/:cohortId').put(editCohort).delete(deleteCohort);
+router.post('/login', login);
+router.get('/logout', logout);
 
-router.put('/projects/:projectId', editProject);
-router.post('/projects', addProject);
+router
+  .route('/cohorts/:cohortId')
+  .put(protectedRoute, editCohort)
+  .delete(protectedRoute, deleteCohort);
```

Comment:
because you use here **router.route()**
you don't need to wite authentication middleware before each method
you can only make it like 
```js
router
  .route('/cohorts/:cohortId')
  .all(protectedRoute)
  .put(editCohort)
  .delete(deleteCohort);
```